src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const {
+  SendTransaction,
+  PartialTxProposal,
+  helpersUtils,
+  PartialTxInputData,
+  PartialTx,
+  txApi,
+  storage,
+  constants: { HATHOR_TOKEN_CONFIG },
+  wallet: oldWallet,
+  dateFormatter,
+} = require('@hathor/wallet-lib');
+const { assembleTransaction } = require('../../../services/atomic-swap.service');
+const { parametersValidation } = require('../../../helpers/validations.helper');
+const { lock, lockTypes } = require('../../../lock');
+const { cantSendTxErrorMessage } = require('../../../helpers/constants');
+const { mapTxReturn } = require('../../../helpers/tx.helper');
+
+async function buildTxProposal(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const sendTokens = req.body.send || { tokens: [] };
+  const receiveTokens = req.body.receive || { tokens: [] };
+  const changeAddress = req.body.change_address || null;
+  const partialTx = req.body.partial_tx || null;
+  let markAsSelected = true;
+  if (req.body.lock !== undefined) {
+    markAsSelected = req.body.lock;
+  }
+  const utxos = [];
+
+  if (sendTokens.tokens.length === 0 && receiveTokens.tokens.length === 0) {
+    res.status(400).json({ success: false, error: 'Should have at least one operation' });
+    return;
+  }
+
+  const proposal = partialTx
+    ? PartialTxProposal.fromPartialTx(partialTx, network) : new PartialTxProposal(network);
+
+  if (sendTokens.utxos && sendTokens.utxos.length > 0) {
+    try {
+      const currentTs = dateFormatter.dateToTimestamp(new Date());
+      for (const utxo of sendTokens.utxos) {
+        const txData = await new Promise((resolve, reject) => {
```

Comment:
I don't think we should do any validations here. The lib code should do everything needed for the integration, we can't expect every use case that will use the lib to implement all this code.

Apart from that, I don't feel we should have this validation at all, I might be preparing a transaction that will be executed in a few hours but I want to start preparing it, so I can get the other person's signature. If the transaction has any problem, the full node will reject it.

Reply:
During our talk we decided to remove the fullnode api call here and get the transaction from the wallet history with `getTx`.
The reason for this is that while this removes the possibility to add utxos from another wallet it removes an external dependency (i.e. call to the fullnode) when building the proposal.